core/commands/partials/vagrant_configuration_generator.rb

```diff
@@ -396,6 +416,20 @@ def generate_node_defenition(node_params, cookbook_path, path)
     end
   end
 
+  # Parse the products lists and path to the products configurations directory from configuration of node.
+  #
+  # @param node [Array] internal name of the machine specified in the template
+  # @return [Array<Hash>, String] list of parameters of products, path to the products configurations directory.
+  def parse_products_info(node)
```

Comment:
I see no need for this method to parse BOTH cnf_template_path and products description. You must provide two separate methods.